server/schedule/hot_cache.go

```diff
@@ -45,38 +45,36 @@ func newHotSpotCache() *HotSpotCache {
 
 // CheckWrite checks the write status, returns whether need update statistics and item.
 func (w *HotSpotCache) CheckWrite(region *core.RegionInfo, stores *core.StoresInfo) (bool, *core.RegionStat) {
-	var WrittenBytesPerSec uint64
+	var (
+		WrittenBytesPerSec uint64
+		value              *core.RegionStat
+	)
 	v, isExist := w.writeFlow.Peek(region.GetId())
 	if isExist && !Simulating {
-		interval := time.Since(v.(*core.RegionStat).LastUpdateTime).Seconds()
+		value = v.(*core.RegionStat)
+		interval := time.Since(value.LastUpdateTime).Seconds()
 		if interval < minHotRegionReportInterval {
 			return false, nil
 		}
 		WrittenBytesPerSec = uint64(float64(region.WrittenBytes) / interval)
 	} else {
 		WrittenBytesPerSec = uint64(float64(region.WrittenBytes) / float64(RegionHeartBeatReportInterval))
 	}
-	region.WrittenBytes = WrittenBytesPerSec
-
-	// hotRegionThreshold is use to pick hot region
-	// suppose the number of the hot Regions is statCacheMaxLen
-	// and we use total written Bytes past storeHeartBeatReportInterval seconds to divide the number of hot Regions
-	// divide 2 because the store reports data about two times than the region record write to rocksdb
-	divisor := float64(statCacheMaxLen) * 2 * storeHeartBeatReportInterval
-	hotRegionThreshold := uint64(float64(stores.TotalWrittenBytes()) / divisor)
 
-	if hotRegionThreshold < hotWriteRegionMinFlowRate {
-		hotRegionThreshold = hotWriteRegionMinFlowRate
-	}
-	return w.isNeedUpdateStatCache(region, hotRegionThreshold, WriteFlow)
+	hotRegionThreshold := calculateWriteHotThreshold(stores)
```

Comment:
missing `region.WrittenBytes = WrittenBytesPerSec`?

Reply:
This is unnecessary.